### PR TITLE
Upgrade Bublé

### PR DESCRIPTION
**Pull request by taion**
I want some of the newer Bublé features like rest property support.

### CODE REVIEW

File: src/utils/transpile/index.js [hunk 1]
```diff
@@ -6,15 +6,12 @@ export const generateElement = (
   { code = '', scope = {} },
   errorCallback
 ) => {
-  // NOTE: Workaround for classes, since buble doesn't allow `return` without a function
-  const transformed = transform(code)
-    .trim()
-    .replace(/^var \w+ =/, '')
-    .replace(/;$/, '')
+  // NOTE: Workaround for classes and arrow functions.
+  const transformed = transform(`(${code})`).trim()
```
Comment (taion) [inline src/utils/transpile/index.js]: Bublé changed how standalone arrow functions were transpiled in https://github.com/Rich-Harris/buble/commit/f2bed343d68232f3dc77aab09555173f64acecb3 in such a way that we can't just take the inline result any more.

This workaround seems to suffice for both standalone arrow functions and for classes. It does add an IIFE for the class case, but I think that's probably okay.

**Comment by alasdairhurst** [comment]
I'd like this. I ran into an issue today regarding JSX properties which were throwing compilation errors: (notice the -lack of- spaces)
```Error: Cannot overwrite a zero-length range – use insertLeft or insertRight instead```
```
	<div className={"foo"}disabled />;
```
```
	<div className="foo"disabled />;
```

Supposedly this is fixed in the latest version (or at least master) https://github.com/Rich-Harris/buble/issues/96

**Comment by taion** [comment]
Ping @philpl

**Comment by philpl** [comment]
@taion Sorry, I haven't gotten around to check this yet. I think the modification you commented on might still cause trouble if it's missing. I'll test it first thing tomorrow.

Thanks in advance for taking the time to look into this! 🙌

**Comment by taion** [comment]
Any updates here? We're blocked on this for getting our live examples to fully work for React-Bootstrap.

**Comment by philpl** [comment]
@taion Sorry for the delay! I must admit, I totally forgot to follow up on this 😢

**Comment by taion** [comment]
Thanks! We'd really appreciate it if you could cut a release or a prerelease with this fix at some point soon.

**Comment by philpl** [comment]
@taion It's already out :wink:

**Comment by taion** [comment]
Thank you!

**Comment by taion** [comment]
@philpl 

I just realized upon closer inspection (checking through other of our examples) that this does change something. Previously if you had something like `<div>Asdf</div>;`, this would work in `generateElement`. Now it has to be `<div>Asdf</div>`.

From my perspective, I don't consider this a regression, since I think the expression with the semicolon is weird and broken, but if it ends up being a problem for others, maybe just removing any trailing semicolon in the input code would be sufficient>?

**Comment by alasdairhurst** [comment]
Hmm that smells like a breaking change then. I know that my project uses semicolons this way (in react-live) due to linting rules. From the sound of things, if i try and build my project npm install will pick up this version and, it will be broken.

**Comment by philpl** [comment]
@taion I wouldn't consider that a regression per se. Let's see if someone reports it in an isolated issue, to see whether it actually affects someones assumptions.

@alasdairhurst It's a breaking change indeed, but one of something that wasn't guaranteed to work, as I wouldn't expect it to work. Your project should still work if you're using `noInline` mode, but if this breaks something in `inline` mode, let's open a new issue and investigate it further :wink:

**Comment by taion** [comment]
Oh, crud, I forgot about linting. Prettier actually makes us insert a semicolon for cases like https://github.com/react-bootstrap/react-bootstrap/blob/v0.32.0/www/src/examples/AlertBasic.js.

**Comment by taion** [comment]
https://github.com/FormidableLabs/react-live/pull/54

Going to verify that this works as expected live first.

**Comment by alasdairhurst** [comment]
@philpl  thanks for pointing out noinline. Missed out on that and it should be useful! Although in this case, only partially.

I externalize all my jsx examples which are gathered together and rendered with react-live, so most of the time i have a file which has the contents "<foo> bar </foo>". Linting is enabled since i may have javascript mixed in. 
```
class Foo extends React.Component {
 render() {
  const { foo } = this.props;
  render (
    <div>
        {foo}
    </div>
  );
 }
}
```

I guess i could work around it by using different file extensions to filter linting rules and inline props passed to react-live (.jsx vs .inline.jsx or something), but ideally it would work as before.

**Comment by taion** [comment]
@alasdairhurst I fixed the regression in https://github.com/FormidableLabs/react-live/pull/54 and I've confirmed that v1.9.2 works for my cases.

**Pull request merged**